services/forms/repo_form.go

```diff
@@ -599,6 +599,32 @@ func (f *MergePullRequestForm) Validate(req *http.Request, errs binding.Errors)
 	return middleware.Validate(errs, ctx.Data, f, ctx.Locale)
 }
 
+// SetDefaults for options who n
+func (f *MergePullRequestForm) SetDefaults(pr *models.PullRequest) {
+	if len(f.Do) == 0 {
+		f.Do = "merge"
+	}
+
+	f.MergeTitleField = strings.TrimSpace(f.MergeTitleField)
+	if len(f.MergeTitleField) == 0 {
+		switch f.Do {
+		case "merge":
+			f.MergeTitleField = pr.GetDefaultMergeMessage()
+		case "rebase-merge":
+			f.MergeTitleField = pr.GetDefaultMergeMessage()
+		case "squash":
+			f.MergeTitleField = pr.GetDefaultSquashMessage()
+
+		}
```

Comment:
The blank line could be removed.

And I think here need a `default` branch to catch unexpected strings, and report errors.

Reply:
no other merge styles dont require a commit message